File: Data Structures & Algorithms/search-2d-matrix/submission-1.cpp
class Solution {
public:
    bool searchMatrix(std::vector<std::vector<int>>& matrix, int target) {
        int n = matrix.size();
        int m = matrix[0].size();
        int left = 0, right = n * m - 1;
        while (left <= right)
        {
            int mid = (left + right) / 2;
            int element = matrix[mid / m][mid % m];
            if (element == target) return true;
            if (element < target) left = mid + 1;
            else right = mid - 1;
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 1;
        int right = numbers.length;
        while (left < right) {
            if (numbers[left-1] + numbers[right-1] == target)
                return new int[]{left, right};
            if (numbers[left-1] + numbers[right-1] < target)
                left++;
            else right--;
        }
        return new int[0];
    }
}